THEMES-119: perf localization

JackHowa opened the pull request:
https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=838&projectKey=THEMES&modal=detail&selectedIssue=THEMES-119&assignee=5e387d6a1b1d910e5dfd7a9b


- maintains same behavior showing locale and timezone of supported timezones specified. see test instructions for changing sites via https://github.com/WPMedia/engine-theme-sdk/pull/255
- we may need to do some messaging on this to give clients more flexibility for timezones
- @badmintonkid mentioned using postinstall to target locales and timezones as needed https://github.com/wapopartners/LeParisien-PageBuilder-Fusion-Features/blob/master/scripts/postinstall.js
- related to Eric's comment here about timezone being about 50% of the bundle https://arcpublishing.atlassian.net/browse/TMEDIA-13?focusedCommentId=525572&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel
- to test, use bundle analyzer https://redirector.arcpublishing.com/alc/arc-products/pagebuilder/fusion/documentation/recipes/webpack-stats.md?version=2.7

before 

![Screen Shot 2021-03-12 at 15 22 46](https://user-images.githubusercontent.com/5950956/111003780-611dde80-834d-11eb-9a30-6ae336a69886.png)

<img width="430" alt="Screen Shot 2021-03-12 at 16 14 15" src="https://user-images.githubusercontent.com/5950956/111004124-033dc680-834e-11eb-8d4a-0346cbfbcb0c.png">

after: 

![Screen Shot 2021-03-12 at 15 46 00](https://user-images.githubusercontent.com/5950956/111003718-477c9700-834d-11eb-8cf2-3c5eac0820d3.png)

![Screen Shot 2021-03-12 at 15 45 55](https://user-images.githubusercontent.com/5950956/111003744-53685900-834d-11eb-8277-859bc59b906e.png)

JackHowa committed:
ci: Lock down package version because not a popular package with minor versions

JackHowa committed:
feat: Factor out local date time helper, add types, include clarifying comment datetime

JackHowa committed:
test: Add clarification that if we didn't match to english this would work with timezone package india

JackHowa committed:
docs: Add todo for country and language comments

JackHowa committed:
docs: Add todo in tests for passing in languages

JackHowa committed:
test: Clarify test goal

JackHowa committed:
Fix comment typo

JackHowa committed:
Merge branch 'THEMES-84-valid-date-format-check' into THEMES-84-localize-date-timezones

JackHowa committed:
perf: Require only what timezone is used per locale

JackHowa committed:
feat: Add locale and require for date time formatting

JackHowa committed:
feat: Fix typo not requiring seoul korea locale

JackHowa committed:
feat: Restore default utc behavior, allow mixmatch locale and timezone

JackHowa committed:
style: Remove unnecessary test comment showing import vs before

brentmiller1973 commented on src/utils/localizeDateHelper.ts:
```diff
@@ -1,21 +1,34 @@
-
+/* eslint-disable @typescript-eslint/no-var-requires,global-require */
 // tz docs https://bigeasy.github.io/timezone/
+const tz = require('timezone')(
```
This.  This is awesome.  Wondering if this pattern should be included in client docs for custom components for like best practices.

brentmiller1973 reviewed (approved):
Great job Jack

brentmiller1973 commented on src/utils/localizeDateHelper.ts:
```diff
@@ -1,21 +1,34 @@
   timeZone: string,
 ): string {
   let locale = null;
-  // locale is the language, like English has January
```
Sort of like the comments, helps for the new dev.  Could they just be updated a bit here?

JackHowa replied:
updated how?

brentmiller1973 reviewed (approved):
Approved, but I do like keeping or updating the comments for new devs that might come onto this project soon.

JackHowa commented:
Will have to update this to get ready for clients. This could be breaking depending on how clients are using this. Also, want to update it for a more scalable solution with postinstall

JackHowa commented:
awaiting Sara's further instructions for which timezones to include

JackHowa commented:
we were previously installing all but only allowing 6 locales 

all locales used
![Screen Shot 2021-03-15 at 14 49 01](https://user-images.githubusercontent.com/5950956/111213881-3e830400-859f-11eb-85f7-df5937820323.png)

JackHowa commented on src/utils/localizeDateHelper.ts:
```diff
@@ -1,21 +1,34 @@
+  require('timezone/sv_SE'),
+  require('timezone/en_US'),
+  require('timezone/de_DE'),
+  require('timezone/fr_FR'),
```
need to add Norwegian via: 

```
Language to use in dates should be restricted to a select menu of only these supported languages: 

English

Spanish

French

German

Korean

Japanese

Swedish

Norwegian

JackHowa replied:
done

JackHowa committed:
feat: Add norwegian and comment names of languages

JackHowa committed:
Merge branch 'canary' into THEMES-84-perf-localization

JackHowa committed:
fix: Updated to include oslo per feature pack

via https://github.com/WPMedia/Fusion-News-Theme/blob/master/blocks.json#L222

JackHowa committed:
style: Remove eslint ignore rule

JackHowa committed:
feat: Add Auckland and other americas timezones

Remove some as well based on list of requireds: 44 America/New_York America/Chicago America/Los_Angeles America/Mexico_City Pacific/Auckland Europe/Oslo Europe/Stockholm Europe/Paris

No longer support Seoul TOkyo  madrid and busingen supported in fusion news theme

JackHowa committed:
style: Reorder require and commented out timezones

JackHowa committed:
test: Fix excluded korean test. Add spanish speaking new zealanders and french-speaking parisian tests

JackHowa committed:
test: Use more informative tests with the am/pm to show times
